src/utils/transaction.ts

```diff
@@ -406,6 +408,207 @@ export const klimaDaoStaking = async (
+        tokenAddress1,
+        amount1,
+        amountMin1,
+        amountMin2,
+        receiverAddress,
+        createTimestamp,
+      ]);
+    }
+
+    console.log("ETHERSPOT::4")
+
+    const encodedData = ContractInterface.encodeFunctionData('addLiquidity', [
+      tokenAddress1,
+      tokenAddress2,
+      amount1,
+      amount2,
+      amountMin1,
+      amountMin2,
+      receiverAddress,
+      createTimestamp,
+    ]);
+
+    const approveAbi = [
+      {
+        constant: false,
+        inputs: [
+          { name: '_to', type: 'address' },
+          { name: '_value', type: 'uint256' },
+        ],
+        name: 'approve',
+        outputs: [{ name: 'result', type: 'bool' }],
+        payable: false,
+        stateMutability: 'nonpayable',
+        type: 'function',
+      },
+    ];
+    console.log("ETHERSPOT::5")
+
+    const ERC20Intance = new ethers.utils.Interface(approveAbi);
+
+    const approvalData1 = ERC20Intance.encodeFunctionData('approve', [contractAddress, amount1]);
+
+    const approvalData2 = ERC20Intance.encodeFunctionData('approve', [contractAddress, amount2]);
+
+    const newEncodeAddLiquidityTransactions = {
+      to: contractAddress,
+      data: encodedData,
+      chainId: CHAIN_ID.XDAI,
+      value: '0',
+      createTimestamp,
+      status: CROSS_CHAIN_ACTION_STATUS.UNSENT,
+    };
+
+    const newApprovalTransaction1 = {
+      to: tokenAddress1,
+      data: approvalData1,
+      chainId: CHAIN_ID.XDAI,
+      value: '0',
+      createTimestamp,
+      status: CROSS_CHAIN_ACTION_STATUS.UNSENT,
+    };
+
+    const newApprovalTransaction2 = {
+      to: tokenAddress2,
+      data: approvalData2,
+      chainId: CHAIN_ID.XDAI,
+      value: '0',
+      createTimestamp,
+      status: CROSS_CHAIN_ACTION_STATUS.UNSENT,
+    };
+
+    let newNativeTokenEndodedData = encodedEthData
+      ? {
+          to: contractAddress,
+          data: encodedEthData,
+          chainId: CHAIN_ID.XDAI,
+          value: '0',
+          createTimestamp,
+          status: CROSS_CHAIN_ACTION_STATUS.UNSENT,
+        }
+      : null;
+
+    const isAnyTokenAddressZero = isZeroAddress(tokenAddress1) || isZeroAddress(tokenAddress2);
+
+    let transactions : any[] = [];
+
+    if (isZeroAddress(tokenAddress1)) {
+      transactions = [newApprovalTransaction2];
+    } else if (isZeroAddress(tokenAddress2)) {
+      transactions = [newApprovalTransaction1];
+    }
+
+    console.log("transactionsXXX", transactions)
+
+    if (isAnyTokenAddressZero && newNativeTokenEndodedData) {
+      transactions = [...transactions, newNativeTokenEndodedData];
+    } else {
+      transactions = [...transactions, newEncodeAddLiquidityTransactions];
+    }
+
+    return { result: { transactions, provider: 'LiFi' } };
+  } catch (error) {
+    console.log('errorPPP', error instanceof Error ? error.message : error);
+    return { errorMessage: 'Failed build transaction 2!' };
```

Comment:
Was the '2' here was for logging purposes? Maybe remove this if the end user sees this error
```suggestion
    return { errorMessage: 'Failed to build transaction!' };
```